.github/workflows/build.yaml

```diff
@@ -0,0 +1,199 @@
+    steps:
+    - name: "Checkout source code at current commit"
+      uses: actions/checkout@v2
+    - name: "Set up Docker Buildx"
+      uses: docker/setup-buildx-action@v1
+    - name: "Build docker image"
+      uses: docker/build-push-action@v2
+      with:
+        tags: cloudposse/docs
+        outputs: type=docker,dest=/tmp/image.tar
+    - name: "Import Image"
+      run: |
+        ls -lhrta /tmp
+        ls -lhrta /tmp/image.tar
+        docker load --input /tmp/image.tar
+    - name: "Build Hugo"
+      run: |
+        set +e -x
+        make lint
+        make release
+        make real-clean hugo/build
+    - name: "Verify that Hugo files were built"
+      run: |
+        ls -lhatr
+        ls -lhatr ./public
+    - name: "Upload Hugo site artifact"
+      uses: actions/upload-artifact@v2
+      with:
+        name: hugo-site
+        path: ./public
+    #hugo-cache-pull:
+    - name: "Get SHAs"
+      id: get_shas
+      run: |
+        echo "::set-output name=current_sha_short::$(git rev-parse --short HEAD)"
+        export PREVIOUS_SHA_SHORT=$(git rev-parse --short HEAD~1)
+        if [ 0 -eq $(git merge-base --is-ancestor $COMMIT_ID HEAD) ]; then
+          echo "::set-output name=previous_sha_short::$(git rev-parse --short HEAD~1)"
+        else
+          echo "::set-output name=previous_sha_short::false"
+        fi
+    - name: "Get cache location"
+      id: get_cache_location
+      run: |
+        [[ "$(cat ${HTMLTEST_CONFIG} | grep OutputDir)" =~ \"(.+)\" ]] && export location="${BASH_REMATCH[1]}"
+        echo "Cache location ${location}"
+        echo "Present directory: $(pwd)"
+        echo "${location}" > ./cache_location.txt
+        echo "::set-output name=cache-location::${location}"
+    - name: "Upload cache location artifact"
+      uses: actions/upload-artifact@v2
+      with:
+        name: cache-location
+        path: ./cache_location.txt
+    - name: "Pull previous htmltest cache, if it exists"
+      id: htmltest-cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.get_cache_location.outputs.cache-location }}
+        key: htmltest-cache-${{ steps.get_shas.outputs.previous_sha_short }}
+    - name: "(Re-)Generate cache"
+      run: make real-clean smoketest || true
+    - name: "Upload Hugo cache artifact"
+      uses: actions/upload-artifact@v2
+      with:
+        name: hugo-cache
+        path: ${{ steps.get_cache_location.outputs.cache-location }}
+    - name: "Identify cache file by inspection"
+      run: |
+        ls -lhatr .
+        ls -lhatr /
+
+  hugo-cache-push:
+    name: "Push Updated Hugo Cache"
+    runs-on: ubuntu-latest
+    needs: hugo-build
+    steps:
+    - name: "Get SHA"
+      id: get_sha
+      run: |
+        echo "::set-output name=current_sha_short::$(git rev-parse --short HEAD)"
+    - name: "Download cache location artifact"
+      uses: actions/download-artifact@v2
+      with:
+        name: cache-location
+        path: ./cache_location.txt
```

Comment:
The cache location seems deterministic to me

Reply:
yeah, you're right. we're using an env now.